### PR TITLE
Portfolio

### PR DESCRIPTION
**Pull request by Najeeba-Qarqin**
Portfolio project created in HTML, CSS and JavaScript and designed by UI/UX designers.

Screenshot 2024-08-04 010547

### CODE REVIEW

**Comment by Najeeba-Qarqin** [comment]
Hi @Hasinarahman👋 
Please review my project.

**Comment by Hasinarahman** [comment]
Hi @Najeeba-Qarqin  🙋‍♀️,
Highlights 🌞
🎉 Great Design Implemented
✔️ Created a professional Repo (You created a repository with your project name)
✔️ Correct use of Gitflow (You created the development branch and made it your default branch, then created and committed your work on your feature branch)
✔️ Commit messages are meaningful
✔️ Setup the linters correctly (You added the correct linters config into your project according to the language you used on your code)
✔️ No linter errors (All linters checks have been passed 🟢)
✔️ You have followed the Personal Portfolio project equirments
✔️ You have followed the all design of figma from UX/UI student
✔️ all of your links works good
✔️ Created the pull request (From your feature branch to the development branch you created a pull request and submitted the pull request link for review)
✔️ Good PR description (You added a meaningful description to your PR about your work and the features you added on this pull request)
✔️Your code is clean
✔️Your repo has a README.md file

just one thing to say your view button of the certificate is so close to the another certificate please give some space on design it.

**Comment by Najeeba-Qarqin** [comment]
Hi @Hasinarahman 
Thank you for your feedback!
I'll implement them.

**Comment by Najeeba-Qarqin** [comment]
@Hasinarahman I implemented your feedback please review again.

**Comment by ahzamir** [comment]
@Najeeba-Qarqin 

- Well done Najeeba, your project is almost done but has some minor issues first of all before I check your project design and functionality, I need to match it with the Figma design so please provide the link to the Figma design in reply to this message and also on your README.md file.

File: script.js [hunk 1]
```diff
@@ -0,0 +1,251 @@
...
+</div>
+</div>
+`;
+const about = document.getElementById('about');
+const skills = {
+  languages: ['HTML', 'CSS', 'javaScript'],
+  tools: ['Git', 'GitHUb'],
+  certification: [
+    { name: 'Web Design', link: 'www.com', image: './images/webDesign.png' },
+    { name: 'JavaScript', link: 'www.com', image: './images/JS.png' },
```
Comment (ahzamir) [inline script.js]: - Add the actual link to your certifications.

File: script.js [hunk 2]
```diff
@@ -0,0 +1,251 @@
...
+    liveLink: 'https://najeeba-qarqin.github.io/ProductLandingPage',
+    sourceLink: 'https://github.com/Najeeba-Qarqin/ProductLandingPage',
+  },
+  {
+    title: 'Portfolio',
+    shortDescription: 'Portfolio project a simple personal portfolio website one of certification projects in freeCodeCamp...',
+    longDescription: 'Portfolio project a simple personal portfolio website one of certification projects in freeCodeCamp Responsive Web Design program build in HTML and designed in CSS.',
+    image: './images/Portfolio.png',
+    technology: ['HTML', 'CSS', 'Git', 'GitHub'],
+    liveLink: 'https://najeeba-qarqin.github.io/Prtfolio',
```
Comment (ahzamir) [inline script.js]: - This link is broken and not working

File: script.js [hunk 3]
```diff
@@ -0,0 +1,251 @@
...
+    liveLink: 'https://najeeba-qarqin.github.io/CashRegister',
+    sourceLink: 'https://github.com/Najeeba-Qarqin/CashRegister',
+  },
+  {
+    title: 'Pokemon Search App',
+    shortDescription: 'Pokemon Search App project a simple app can show data and information about any pokemon from their ID ...',
+    longDescription: 'Pokemon Search App project a simple app can show data and information about any pokemon from their ID or name, one of certification projects in freeCodeCamp Algorithms And Data Structure program build in HTML, CSS and JavaScript.',
+    image: './images/PokemonSearch.png',
+    technology: ['HTML', 'CSS', 'Git', 'GitHub'],
+    liveLink: 'https://najeeba-qarqin.github.io/Prtfolio',
```
Comment (ahzamir) [inline script.js]: - This link is broken, please add the actual live link for this project.

File: index.html [hunk 4]
```diff
@@ -0,0 +1,170 @@
...
+    </div>
+  </section>
+
+  <section id="contact" class="flex">
+    <h2 class="span">Contact Me</h2>
+    <div class="contact flex">
+      <p class="para">If you have an application you are interested in developing, a feature that you need builr or a
+        project that need coding. I'd love to help with it. Or if you have any feedback and message please share with
+        me.</p>
+      <form action="" name="myform" class="myform flex" id="myform" method="get">
```
Comment (ahzamir) [inline index.html]: - You have to use [Formspree](https://formspree.io/) for you contact form submission action.
  - You can check the project's requirements and the documentation of the formspree for more details.

File: script.js [hunk 5]
```diff
@@ -0,0 +1,251 @@
...
+
+const form = document.getElementById('myform');
+const name = document.getElementById('name');
+const email = document.getElementById('email');
+const message = document.getElementById('message');
+const nameError = document.getElementById('name_error');
+const emailError = document.getElementById('email_error');
+const messageError = document.getElementById('message-error');
+form.addEventListener('submit', (e) => {
+  const emailChecker = /^[\w-]+@[a-zA-Z\d-]+\.[a-zA-Z]{2,}$/;
```
Comment (ahzamir) [inline script.js]: - You have to check if the user writes his email in capital or small letters and if it has any capital letters you have to show an error message:
  - Check the project requirements for more details.

File: README.md [hunk 6]
```diff
@@ -1 +1,122 @@
...
+# Portfolio 🚀
+
+## Project Description 📝
+
+> This is my portfolio project i created using HTML, CSS and javaScript it's a responsive webpage.
+
+Examples:
+
+- I learned how to make dynamic webpage using js.
+- You can give some information about me.
```
Comment (ahzamir) [inline README.md]: - Solve this, as the user does not give information about you they will read info about you.

File: README.md [hunk 7]
```diff
@@ -1 +1,122 @@
...
+  `;
+  popup.classList.remove('hidden');
+};
+
+popupBtn.addEventListener('click', () => {
+  popup.classList.add('hidden');
+});
+
+
+```
```
Comment (ahzamir) [inline README.md]: - You can remove this part.

File: README.md [hunk 8]
```diff
@@ -1 +1,122 @@
...
+## Demo 📸
+
+
+![Demo](/images/DEMO.png)
```
Comment (ahzamir) [inline README.md]: - The demo image is not showing up as your image is `Demo.png` not `DEMO.png`

File: README.md [hunk 9]
```diff
@@ -1 +1,122 @@
...
+
+
+### Live Link
+
+[The_live_link_of_this_project](https://najeeba-qarqin.github.io/Portfolio-JS/)
```
Comment (ahzamir) [inline README.md:22]: - This link is not working, but no worries you can update it after you deployed this project

File: README.md [hunk 10]
```diff
@@ -1 +1,122 @@
...
+
+## Technologies Used 🛠️
+
+
+- HTML
+- CSS
+- JS
+
```
Comment (ahzamir) [inline README.md]: - Change this to Technologies and Tools used and then also add Figma on the below list

File: README.md [hunk 11]
```diff
@@ -1 +1,122 @@
...
+
+- LinkedIn: [Najeeba Qarqin](https://www.linkedin.com/in/najeeba-qarqin-5419502ab?utm_source=share&utm_campaign=share_via&utm_content=profile&utm_medium=android_app)
+- Email: [Najeeba Qarqin](najeebaqarqin@gmail.com)
+- GitHub: [Najeeba Qarqin](https://github.com/Najeeba-Qarqin)
+- UI/UX Designers:
+  ####  Maryam Yousufi
+  ####  Sara Shirzad
+  ####  Sadaf Ahmadi
+  ####  Farkhunda Raufi
```
Comment (ahzamir) [inline README.md]: - Add a link to the Figma design here and also you need to add at least one social media link for these UI/UX designers.

**Review by ahzamir — CHANGES_REQUESTED** [review]
Hey @Najeeba-Qarqin , Well done and great job by implementing most of the requirements:

- Well done your project's functionality almost meets most of the requirements, but still your project has some minor issues you need to work on.

- This review and feedback is just for your project's README.md, HTML, and JavaScript requirements. In order to get feedback and review for your design you have to provide link of the Figma design on your readme file.

**Comment by Najeeba-Qarqin** [comment]
> @Najeeba-Qarqin
> 
> * Well done Najeeba, your project is almost done but has some minor issues first of all before I check your project design and functionality, I need to match it with the Figma design so please provide the link to the Figma design in reply to this message and also on your README.md file.

@ahzamir  Hi sir thank you for feedback 
Ok this is the link: 
 https://www.figma.com/invites/claim/M4WghDpaGEamietPGHcEHoCfP4GsHkX0yXRl13g9J5TR?fuid=1379424393568974635

File: README.md (hunk 7 above)
Comment (Najeeba-Qarqin) [reply README.md:93]: Why sir, and how??🥺

Comment (Najeeba-Qarqin) [reply README.md]: Understand 🙂

**Comment by Najeeba-Qarqin** [comment]
@ahzamir Hi sir!
I implemented  your feedback can you please check the project again.

File: script.js [hunk 12]
```diff
@@ -0,0 +1,255 @@
...
+  } else {
+    emailError.innerText = '';
+  }
+  if (message.value.length <= 12) {
+    e.preventDefault();
+    messageError.innerHTML = 'Please write a message!';
+  } else {
+    messageError.innerHTML = '';
+  }
+});
```
Comment (ahzamir) [inline script.js]: @Najeeba-Qarqin 

- Well done, just one thing else you need to implement here after the user submits the form you need to clear the Form fields inner text and the LocalStorage Data.

File: styles.css
Comment (ahzamir) [inline styles.css:1]: - The background color for the project pages including the project header is a linear gradient color, you have to use the exact linear gradient color on your project background, please have a look at the images to see the changes between your design and the Figma design:
- The buttons' background color also has little gradient and shadow on the Figma Design.
- The padding and sizing of the Elements on the Home page are a little bit different from the Figma design

- The Figma Design:
image
|
image


- Your Design:
  - Desktop version:
  - Here I added just the desktop version design as a reference but implement these changes for both desktop and mobile versions.
  
image
|
image

Comment (ahzamir) [inline styles.css:1]: - The placing od the button on the about page is different from the Figma design:
- Same issue on the About page, the button design and sizing are not aligned and the same as the Figma design:

- Your Design:
image

image

- The Figma Design:
image
image

Comment (ahzamir) [inline styles.css:1]: - Padding issue on the skills page, please check the Figma design and add padding to your elements according to the Figma Design:
- Also the text under skills should be centered

- Your Design Mobile version:
image

- Figma Design:
image

Comment (ahzamir) [inline styles.css:1]: @Najeeba-Qarqin 

- Although I accept that the changes you added is looking more professional and more cool on your project, but as this project is a lesson for you to follow the exact design which is on the Figma so you need to design exactly like the Figma Design:
- No background for the projects card:

- Your design:
image

- Figma Design:
image

Comment (ahzamir) [inline styles.css:1]: - These button icons imported with a very low format graphic although it's correct on the Figma solve this:
image

**Review by ahzamir — CHANGES_REQUESTED** [review]
@Najeeba-Qarqin 

- Well done Najeeba by implementing the feedback and creating this amazing project just some minor changes on the design of the project are remaining you need to work on:

File: styles.css
Comment (Najeeba-Qarqin) [reply styles.css:1]: > * The background color for the project pages including the project header is a linear gradient color, you have to use the exact linear gradient color on your project background, please have a look at the images to see the changes between your design and the Figma design:
> * The buttons' background color also has little gradient and shadow on the Figma Design.
> * The padding and sizing of the Elements on the Home page are a little bit different from the Figma design
> * The Figma Design:
>   image
>   |
>   image
> * Your Design:
>   
>   * Desktop version:
>   * Here I added just the desktop version design as a reference but implement these changes for both desktop and mobile versions.
> 
> image | image

@ahzamir 
in the body the linear gradient color is for it's border sir not for background i looked at it have only one color for background if i add the linear gradient it make sense like light color.

**Comment by Najeeba-Qarqin** [comment]
@ahzamir Salam sir!
I changed the required points of design that you mentioned just about the background of the body please check again cause when i added linear gradient it makes big difference.

File: styles.css
Comment (ahzamir) [reply styles.css:1]: - From my point of view, I think the body background is a linear gradient consisting of 3 colors, check the Figma design there you can find which colors with which percentages!

**Comment by ahzamir** [comment]
> @ahzamir Salam sir! I changed the required points of design that you mentioned just about the background of the body please check again cause when i added linear gradient it makes big difference.

I checked it, it's a linear gradient background.

**Comment by Najeeba-Qarqin** [comment]
> > @ahzamir Salam sir! I changed the required points of design that you mentioned just about the background of the body please check again cause when i added linear gradient it makes big difference.
> 
> I checked it, it's a linear gradient background.

Screenshot_20240801-195922_Chrome
Sir can you explain please it's just one color for background 🥺

**Comment by Najeeba-Qarqin** [comment]
@ahzamir 
Hi sir!
I added a radial gradient color for background of project.

**Review by ahzamir — APPROVED** [review]
### Hi @Najeeba-Qarqin 🙋‍♀️,

> Excellent work! 💯 You have done a great job implementing all the project requirements. 🥳 Your feature is complete! Congratulations!!!🎊

- **[REQUIRED]** Share your project on this [GROUP](https://www.facebook.com/groups/kick-startdev), add a simple Proud Moment text and a screenshot of your project then besides that add your project link at the end of the text as a GitHub Source code reference and post.

- Share your Project on your LinkedIn account and add a proud moment text with a screenshot.

- Well Done great design Implemented. 🎯🎉👍
image

# Status: Approved ✅
There is nothing to say other than to merge your branch. 🎉🎯✅👍😊🥳

WinkThumbsUpGIF


 ## Highlights 🌞

🎉 Great Design Implemented 
✔️ Created a professional Repo (You created a repository with your project name)
✔️ Correct use of Gitflow (You created the development branch and made it your default branch, then created and committed your work on your feature branch)
✔️ Commit messages are meaningful
✔️ Setup the linters correctly (You added the correct linters config into your project according to the language you used on your code)
✔️ No linter errors (All linters checks have been passed 🟢)✔️ You have followed the Cash Register project task and committed using the command line
✔️ Created the pull request (From your feature branch to the development branch you created a pull request and submitted the pull request link for review)
✔️ Good PR description (You added a meaningful description to your PR about your work and the features you added on this pull request)
✔️Your code is clean
✔️Your repo has a professional README.md file
✔️ Your partner has reviewed your repo
✔️Your design is responsive
✔️Your design is according to the Figma design and even better than that. 🎯

**Cheers 🥂 and Happy coding!!! 👯 Feel free to leave any questions or comments in the PR thread if something is not 100% clear. Remember to tag me @ahzamir in your question so I can receive the notification.**

**Comment by ahzamir** [comment]
@Najeeba-Qarqin 

- Don't forget to:
  - Merge
  - Deploy
  - Share on FB Group & Social Media

> These are part of the requirments

**Pull request merged**